### DISCUSSION TITLE
Advanced table ?

### DISCUSSION THREAD

**Discussion by FabienPoirierPrincipia** [beeware/toga#2780] [Q&A]
I'd like to know if it is possible to use a table with some way to sort data by column and also to filter content ?
By reading the doc, I don't think so but is it possible to extend the behavior of the existing table?
In my Android app, I have a column which contains the type of data and it could be nice to be able to select only some types to display the according data. I could do it with a dedicated form near the table, but it won't be the same user experience.

**Comment by freakboy3742** [comment]
The Toga table API doesn't currently have any built-in mechanism to control sort order by column. If you want to change the sort order of a table, you'd need to do that at the level of the data source (i.e., change the order that the source returns data); but there isn't currently any signal that would let you catch clicks on a table header, for example. If you want this sort of behavior, you'd need to implement platform-specific hooks.

In addition to this, the Table widget on Android should be considered beta. It doesn't expose a UI that is particularly good for mobile devices, and is known to have major performance issues with any table with a large amount of data. If you're planning to build an app that uses the table widget on Android, I'd expect to see some fairly major UI changes in the not-too-distant future.

**Reply by FabienPoirierPrincipia** [reply]
OK. I will do as it is now.
I don't know how to implement platform specific hook. But noted the status on Android.
May be in a future version it could be possible to get a user to catch user action on the header. 
Thanks for your reply.
